sketchformat/prototype.py

```diff
@@ -31,8 +31,9 @@ class FlowOverlaySettings:
     overlayType: int = 0
 
     @staticmethod
-    def Positioned(position: str) -> "FlowOverlaySettings":
+    def Positioned(position: str, manual_offset: Optional[DictXY] = None) -> "FlowOverlaySettings":
         anchor = Point(0.5, 0.5)
+        offset = Point(0, 0) if manual_offset is None else Point.from_dict(manual_offset)
```

Comment:
Why not:
- Set `manual_offset: Point = Point(0, 0)`
- Remove this line
- Undo the changes to DictXY and Point
- In the call to Positioned, pass `Point.from_dict(action.get("overlayRelativePosition"))`

This way we expose less internals to the model, and work with model objects all around. As it is now, you are passing a fig_node dictionary straight to sketchformat which works, but is a bit ugly.

Reply:
I didn't want to handle the Point conversion out of this method because we also need to account for `"overlayRelativePosition"` not being present (which will return `None` at the `get` call and `Point.from_dict` can't handle it. That's why I wanted to keep the Point details inside, because any potential call would have to repeat the same checks.

But I agree this solution is ugly as hell so not sure what to do 🤔

Reply:
Another option would undo the changes in DictXY and expect an `Options[Dict]` and do:

```python
offset = Point(0, 0) if manual_offset is None else Point(manual_offset["x"], manual_offset["y"])
```

Since we know that if `overlayRelativePosition` exists it's a dict with those 2 properties. But that's why I though DictXY would be better. Also, this new alternative would mean to _pollute_ the `Positioned` method with details of the .fig model.

So... which is the lesser evil here? 😅

Reply:
Oh right. Then, what about putting the null check in the caller? You can just do
```python
offset = Point.from_dict(action["overlayRelativePosition"]) if "overlayRelativePosition" in action else None
return FlowOverlaySettings(overlayAnchor=anchor, sourceAnchor=anchor, offset=offset)
```

Reply:
https://github.com/sketch-hq/fig2sketch/pull/50/commits/0e0599fc5cc4ec0fbec0ae68508bcc0c1156d188